filestore/src/main/java/com/yahoo/ycsb/db/FileStoreClient.java

```diff
@@ -0,0 +1,233 @@
+   */
+  @Override
+  public Status read(String table, String key, Set<String> fields, Map<String, ByteIterator> result) {
+    String filename = getDatabaseFileName(table, key);
+
+    try (JsonReader jsonReader = new JsonReader(new FileReader(filename))) {
+      Map<String, ByteIterator> values = gson.fromJson(jsonReader, valuesType);
+
+      for (String field : fields) {
+        result.put(field, values.get(field));
+      }
+
+      return Status.OK;
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    return Status.NOT_FOUND;
+  }
+
+  @Override
+  public Status scan(String table,
+                     String startkey,
+                     int recordcount,
+                     Set<String> fields,
+                     Vector<HashMap<String, ByteIterator>> result) {
+    String filename = getDatabaseFileName(table, startkey);
+
+    try (JsonReader jsonReader = new JsonReader(new FileReader(filename))) {
+      Map<String, ByteIterator> values = gson.fromJson(jsonReader, valuesType);
+      result.add(convertToHashMap(values, fields));
+
+      return Status.OK;
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    return Status.ERROR;
+  }
+
+  @Override
+  public Status update(String table, String key, Map<String, ByteIterator> values) {
+    String filename = getDatabaseFileName(table, key);
+
+    try (JsonReader jsonReader = new JsonReader(new FileReader(filename))) {
+      Map<String, ByteIterator> map = gson.fromJson(jsonReader, valuesType);
+
+      for (String valuesKey : values.keySet()) {
+        map.put(valuesKey, values.get(valuesKey));
+      }
+
+      insert(table, key, map);
+
+      return Status.OK;
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    return Status.ERROR;
+  }
+
+  @Override
+  public Status insert(String table, String key, Map<String, ByteIterator> values) {
+    String filename = getDatabaseFileName(table, key);
+    String output = gson.toJson(values, valuesType);
+
+    try (FileWriter fileWriter = new FileWriter(filename)) {
+      fileWriter.write(output);
+      return Status.OK;
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+
+    return Status.ERROR;
+  }
+
+  @Override
+  public Status delete(String table, String key) {
+    String filename = getDatabaseFileName(table, key);
+
+    if (new File(filename).delete()) {
+      return Status.OK;
+    }
+
+    return Status.ERROR;
+  }
+
+  private <V> HashMap<String, V> convertToHashMap(Map<String, V> map, Set<String> fields) {
+    HashMap<String, V> result = new HashMap<>();
+
+    if (fields != null && fields.size() > 0) {
+      for (String field : fields) {
+        result.put(field, map.get(field));
+      }
+    } else {
+      result.putAll(map);
+    }
+
+    return result;
+  }
+
+  private String getDatabaseFileName(String table, String key) {
+    return outputDirectory + table + "_" + key + ".json";
```

Comment:
How do I ensure that when I load these files into a datastore I'm doing it in the same order that YCSB generated them?